### DISCUSSION TITLE
Tag input on top of the combobox component

### DISCUSSION THREAD

**Discussion by marcospassos** [ariakit/ariakit#1457] [Questions] [answered]
I'm playing with the Combobox component to build a tag input. [Implementing a naive component is straightforward](https://codesandbox.io/s/objective-phoebe-tj4o3l), except that I found no way to hook into the combobox to react upon an option is selected. I could listen to the changes (`setValue`) and check if the value is one of the values in the combobox list, but I bet there is a better way to achieve it.

These are missing parts:

- Whenever the user selects an option, call the `onSelectTag` and clear the input.
-  Hide the popup when there are no matches (just rendering it conditionally would break animations)

**Comment by diegohaz** [comment, marked as answer]
> * Whenever the user selects an option, call the `onSelectTag` and clear the input.

I have this experiment from a few months ago: https://codesandbox.io/s/ariakit-multi-select-combobox-with-tags-yuem9x?file=/src/App.js

The code needs to be updated to the latest version of the `combobox-multiple` example, and there are probably other improvements that need to be made there. But that's the general idea. It also supports arrow key navigation between the input and the tags, and deleting the tags with backspace. We'll eventually have that example on the site.

> * Hide the popup when there are no matches (just rendering it conditionally would break animations)

You can see how it's done here: https://ariakit.org/examples/combobox-textarea (line 31-33)
